share/functions/fish_vi_cursor.fish

```diff
@@ -1,9 +1,4 @@
 function fish_vi_cursor -d 'Set cursor shape for different vi modes'
-    # If we're not interactive, there is effectively no bind mode.
-    if not status is-interactive
-        return
-    end
-
```

Comment:
Nice find - however this check:

https://github.com/fish-shell/fish-shell/blob/6361362996bc6d915a22f010720aefa2f451500e/share/functions/fish_vi_cursor.fish#L2-L5

was added in 39905d41354358471c41b0f5de2d043f204ca0db to avoid the case of emitting an escape sequence for `scp` and similar non-interactive cases. So removing that check seems likely to cause regressions.

I'm not sure what to do. Maybe we could check if stdin is a tty instead?

Reply:
I tested the proposed change and it seems to work, so I pushed it.